exercicios/para-casa/sistema_de_alunas.py

```diff
@@ -1,59 +1,140 @@
+                break
+            except ValueError:
+                print('Erro em processar o valor. Insira um número válido')
     
+    return notas
+
+def obter_presença():
+    quantidade_aulas = int(input('Digite quantas aulas houveram: '))
+    lista_presença = []
+
+    for contador in range(quantidade_aulas):
+        while True:
+            entrada = input(f'Presença na aula #{contador +1} (True/False): ')
+            if entrada in ['True', 'False']:
+                lista_presença.append(eval(entrada))
+                break
+            else:
+                print('Informação inválida. Insira apenas True ou False')
+
+    return lista_presença
+
+def salvar_dados(nome, sobrenome, turma, lista_presença, nota_participação, notas):
+    chave = (nome, sobrenome)
+    dataset[chave] = { 
+        "Turma": turma,
+        "Notas": notas,
+        "Presença": lista_presença,
+        "Participação": nota_participação
+    }
+    print(f'Os dados da aluna {nome} {sobrenome} foram inseridos com sucesso.')
+
 def consultar_lista_alunas():
-    pass
-    #TODO - Implentar a função
-    
+    if dataset:
+        print('Lista de todas as alunas')
+        for chave in dataset.keys():
+            print(f'Nome: {chave[0]} {chave[1]}')
+    else:
+        print('Não há novos cadastros')
+
 def consultar_faltas_aluna():
-    pass
-    #TODO - Implentar a função
-    
+    nome = input('Informe o primeiro nome da aluna: ').title()
+    sobrenome = input('Informe o sobrenome da aluna: ').title()
+    chave = (nome, sobrenome)
+    if chave in dataset:
+        faltas = dataset[chave]["Presença"].count(False)
+        print(f'A aluna {nome} {sobrenome} está com {faltas} faltas')
+    else:
+        print('Aluna não encontrada.')
+
 def consultar_notas_aluna():
-    pass
-    #TODO - Implentar a função
-    
+    nome = input('Informe o primeiro nome da aluna: ').title()
+    sobrenome = input('Informe o sobrenome da aluna: ').title()
+    chave = (nome, sobrenome)
+    if chave in dataset:
+        notas = dataset[chave]["Notas"]
+        print(f'A aluna {nome} {sobrenome} está com as notas: {notas}')
+    else:
+        print('Aluna não encontrada.')
+
 def consultar_status_aprovacao():
-    pass
-    #TODO - Implentar a função
-    
+    nome = input('Insira o primeiro nome da aluna: ').title()
+    sobrenome = input('Insira o sobrenome da aluna: ').title()
+    chave = (nome, sobrenome)
+    if chave in dataset:
+        dados = dataset[chave]
+        media_notas = sum(dados["Notas"]) / len(dados["Notas"])
+        presença = dados["Presença"].count(True) / len(dados["Presença"])
+        participação = dados["Participação"]
+
+        if media_notas >= 6 and presença >= 0.8 and participação > 6:
+            status = 'Aprovada'
+        else:
+            status = 'Reprovada'
+        print(f'A aluna {nome} {sobrenome} está {status} com média final: {media_notas:.2f}')
+    else:
+        print('Aluna não encontrada')
```

Comment:
Ótima implementação do sistema, cuidado com o nome das variáveis para utilizar nomes que sejam mais intuitivos, por exemplo: "dados" poderia ser "dados_aluna", quando for refletir sobre o nome, nós pensamos se o nome em si responde o que ela carrega.

Futuramente, se quiser complementar mais o sistema, você pode pensar em encapsular as funcionalidades, dando mais especifidade a função e dando responsabilidades únicas para ela.

Parabéns pelo trabalho ♥